src/main/java/net/minestom/server/Configuration.java

```diff
@@ -0,0 +1,53 @@
+package net.minestom.server;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.minestom.server.registry.dynamic.chat.ChatDecoration;
+import net.minestom.server.registry.dynamic.chat.ChatType;
+import net.minestom.server.registry.dynamic.chat.ChatTypeBuilder;
+
+public record Configuration(boolean requireValidPlayerPublicKey, Component missingPlayerPublicKeyMessage,
+                            Component invalidPlayerPublicKeyMessage, ChatTypeBuilder playerChatType,
+                            ChatTypeBuilder systemChatType) {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private boolean requireValidPlayerPublicKey = false;
+        private Component missingPlayerPublicKeyMessage = Component.text("Missing public key!", NamedTextColor.RED);
+        private Component invalidPlayerPublicKeyMessage = Component.text("Invalid public key!", NamedTextColor.RED);
+        private ChatTypeBuilder playerChatType = ChatTypeBuilder.builder(ChatType.CHAT.key()).chat(ChatDecoration.contentWithSender("chat.type.text"));
+        private ChatTypeBuilder systemChatType = ChatTypeBuilder.builder(ChatType.SYSTEM.key()).chat();
+
+        public Builder setRequireValidPlayerPublicKey(boolean requireValidPlayerPublicKey) {
```

Comment:
Don't most of the other builders in Minestom omit the `set` prefix?

Reply:
I'll change that, and this config change will be made in master first